File: z-clients/python/diom/apis/idempotency_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    IdempotencyConfigureNamespaceIn,
    IdempotencyConfigureNamespaceOut,
    IdempotencyGetNamespaceIn,
    IdempotencyGetNamespaceOut,
)


class IdempotencyNamespaceAsync(ApiBase):
    async def configure(
        self,
        idempotency_configure_namespace_in: IdempotencyConfigureNamespaceIn,
    ) -> IdempotencyConfigureNamespaceOut:
        """Configure idempotency namespace"""
        body = idempotency_configure_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.idempotency.namespace.configure",
            body=body,
            response_type=IdempotencyConfigureNamespaceOut,
        )

    async def get(
        self,
        idempotency_get_namespace_in: IdempotencyGetNamespaceIn,
    ) -> IdempotencyGetNamespaceOut:
        """Get idempotency namespace"""
        body = idempotency_get_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.idempotency.namespace.get",
            body=body,
            response_type=IdempotencyGetNamespaceOut,
        )


class IdempotencyNamespace(ApiBase):
    def configure(
        self,
        idempotency_configure_namespace_in: IdempotencyConfigureNamespaceIn,
    ) -> IdempotencyConfigureNamespaceOut:
        """Configure idempotency namespace"""
        body = idempotency_configure_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.idempotency.namespace.configure",
            body=body,
            response_type=IdempotencyConfigureNamespaceOut,
        )

    def get(
        self,
        idempotency_get_namespace_in: IdempotencyGetNamespaceIn,
    ) -> IdempotencyGetNamespaceOut:
        """Get idempotency namespace"""
        body = idempotency_get_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.idempotency.namespace.get",
            body=body,
            response_type=IdempotencyGetNamespaceOut,
        )

File: z-clients/javascript/src/apis/idempotencyNamespace.ts
// this file is @generated

import {
    type IdempotencyConfigureNamespaceIn,
    IdempotencyConfigureNamespaceInSerializer,
} from '../models/idempotencyConfigureNamespaceIn';
import {
    type IdempotencyConfigureNamespaceOut,
    IdempotencyConfigureNamespaceOutSerializer,
} from '../models/idempotencyConfigureNamespaceOut';
import {
    type IdempotencyGetNamespaceIn,
    IdempotencyGetNamespaceInSerializer,
} from '../models/idempotencyGetNamespaceIn';
import {
    type IdempotencyGetNamespaceOut,
    IdempotencyGetNamespaceOutSerializer,
} from '../models/idempotencyGetNamespaceOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class IdempotencyNamespace {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Configure idempotency namespace */
    public configure(
        idempotencyConfigureNamespaceIn: IdempotencyConfigureNamespaceIn,
    ): Promise<IdempotencyConfigureNamespaceOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.idempotency.namespace.configure");

        request.setBody(
            IdempotencyConfigureNamespaceInSerializer._toJsonObject(idempotencyConfigureNamespaceIn)
        );
        
        return request.send(
            this.requestCtx,
            IdempotencyConfigureNamespaceOutSerializer._fromJsonObject,
        );
    }/** Get idempotency namespace */
    public get(
        idempotencyGetNamespaceIn: IdempotencyGetNamespaceIn,
    ): Promise<IdempotencyGetNamespaceOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.idempotency.namespace.get");

        request.setBody(
            IdempotencyGetNamespaceInSerializer._toJsonObject(idempotencyGetNamespaceIn)
        );
        
        return request.send(
            this.requestCtx,
            IdempotencyGetNamespaceOutSerializer._fromJsonObject,
        );
    }
}

